### PR TITLE
fix loss = ()

### PR DESCRIPTION
**Pull request by hnyu**
Sometimes we have loss=() and gradient update will be skipped.

### CODE REVIEW

File: alf/algorithms/algorithm.py [hunk 1]
```diff
@@ -1208,8 +1208,10 @@ def update_with_gradient(self,
 
         loss_info = self._aggregate_loss(loss_info, valid_masks, batch_info)
 
-        all_params, gns = self._backward_and_gradient_update(loss_info.loss *
-                                                             weight)
+        loss = loss_info.loss
+        if loss != ():
+            loss = loss * weight
+        all_params, gns = self._backward_and_gradient_update(loss)
```
Comment (emailweixu) [inline alf/algorithms/algorithm.py]: should we skip _backward_and_gradient_update() in this case?

Comment (emailweixu) [reply alf/algorithms/algorithm.py]: Maybe warning_once for this too.

**Review by emailweixu — APPROVED** [review]
